### PR TITLE
executor: add more information in error message of replace statement

### PR DESCRIPTION
**Pull request by jackysp**
### What problem does this PR solve? 
Add more information in error message of replace statement. e.g. number of rows, the index of all rows, string format of error row datum.

### What is changed and how it works?
User will get an error message like,
```
[replace] the 2th of total 2 rows, handle 1: [kv:2]Error: key not exist
```
and the error log will look like,
```
2018/08/15 20:26:52.580 write.go:1842: [error] [replace] the 2th of total 2 rows, handle 1: [kv:2]Error: key not exist, row: [{1 0 0 0 1 [] <nil>} {1 0 0 0 1 [] <nil>}]
2018/08/15 20:26:52.580 tidb.go:183: [info] RollbackTxn for ddl/autocommit error.
2018/08/15 20:26:52.581 session.go:753: [warning] con:1 session error:
[kv:2]Error: key not exist
/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/kv/union_store.go:186: 
/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/store/tikv/txn.go:90: 
/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/session/txn.go:144: 
/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/table/tables/tables.go:520: 
/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/table/tables/tables.go:577: 
/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/executor/write.go:1841: [replace] the 2th of total 2 rows, handle 1
/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/executor/write.go:1893: 
/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/executor/adapter.go:268: 
/Users/yusp/work/goprojects/src/github.com/pingcap/tidb/session/tidb.go:200: 
{
  "currDBName": "test",
  "id": 1,
  "status": 2,
  "strictMode": true,
  "user": null
}
```

### Check List 

Tests 

 - Unit test

PTAL @shenli @coocood

### CODE REVIEW

File: executor/write.go [hunk 1]
```diff
@@ -1838,7 +1838,9 @@ func (e *ReplaceExec) exec(ctx context.Context, rows [][]types.Datum) (types.Dat
 		}
 		oldRow, err1 := e.Table.Row(e.ctx, h)
 		if err1 != nil {
-			return nil, errors.Trace(err1)
+			rErr := errors.Annotatef(err1, "[replace] the %dth of total %d rows, handle %d", idx+1, rowsLen, h)
+			log.Errorf("%s, row: %v", rErr.Error(), row)
```
Comment (shenli) [inline executor/write.go]: > 2018/08/15 20:26:52.580 write.go:1842: [error] [replace] the 2th of total 2 rows, handle 1: [kv:2]Error: key not exist, row: [{1 0 0 0 1 [] <nil>} {1 0 0 0 1 [] <nil>}]

The row format is not friendly for reading.

**Comment by coocood** [comment]
@jackysp 
What's the original PR for this cherry-pick?

File: executor/write.go (hunk 1 above)
Comment (shenli) [reply executor/write.go]: This PR is very small. Why not fix it in this PR?

Comment (coocood) [reply executor/write.go]: Will this error be logged twice?

Comment (jackysp) [reply executor/write.go]: The row info will be only shown in the log file.

**Comment by coocood** [comment]
LGTM

**Comment by jackysp** [comment]
/run-all-tests tikv=release-2.0 pd=release-2.0 tidb-test=release-2.0

**Pull request merged**